src/main/scala/zio/prelude/NonEmptyList.scala

```diff
@@ -54,6 +54,16 @@ sealed trait NonEmptyList[+A] { self =>
   final def count(f: A => Boolean): Int =
     foldLeft(0)((n, a) => if (f(a)) n + 1 else n)
 
+  /**
+   * Removes duplicate elements from this `NonEmptyList`.
+   */
+  final def distinct(implicit A: Hash[A]): NonEmptyList[A] =
+    reduceMapLeft(a => (single(a), Set(a.hash))) {
+      case ((as, seen), a) =>
+        val hash = a.hash
+        if (seen(hash)) (as, seen) else (cons(a, as), seen + hash)
```

Comment:
Hmm what if there is a hash collision? 🤔

Reply:
If hash collisions are a concern, a `Equal` constraint could be used instead. @adamgfraser let me know if you want me to raise a PR.